README.md

```diff
@@ -51,7 +51,11 @@ At the top level, the document definitions object contains a property for each d
 
 #### Document type definitions
 
-Each document type is specified as an object with the following properties:
+Each document type is defined as an object with a number of properties that control authorization, content validation and access control.
+
+##### Essential document properties:
```

Comment:
In addition to documenting the usage of dynamic property/element constraints, I've also reorganized this part of the document into subsections that should help with readability and discoverability. As part of that, the descriptions of a few document-level properties (e.g. `propertyValidators`, `immutable`, etc.) have been moved around in the list but otherwise have not been modified.